### PR TITLE
Catch delta table not found error

### PR DESCRIPTION
**Pull request by milocress**
when the delta table isn't found, surface this error to the user.

### CODE REVIEW

File: llmfoundry/utils/exceptions.py [hunk 1]
```diff
@@ -532,3 +534,18 @@ class UCNotEnabledError(UserError):
...
         message = f'Unity Catalog is not enabled on your cluster.'
         super().__init__(message)
+
+
+class DeltaTableNotFoundError(UserError):
+    """Error thrown when the delta table passed in training does not exist or is not found."""
+
+    def __init__(self, message: str) -> None:
+        self.message = message
+        super().__init__(message)
```
Comment (jjanezhang) [inline llmfoundry/utils/exceptions.py:544]: Can you pass me the table name so I can construct a good message for it in mcloud? or we could pass this message directly to the user if you feel it is perfect for the user.

Comment (milocress) [reply llmfoundry/utils/exceptions.py:544]: The current message is:

```
Error processing `catalog`.`volume_name`.`table_name`: [DELTA_TABLE_NOT_FOUND] Delta table `volume_name`.`table_name` doesn't exist.
```

which I think is pretty good.

**Comment by milocress** [comment]
The `docformatter` hook runs without error locally, it grates like sandpaper against my soul that this hook would run differently on the server.

File: llmfoundry/utils/exceptions.py (hunk 1 above)
Comment (dakinggg) [reply llmfoundry/utils/exceptions.py:544]: I think @jjanezhang wanted the table name as an attr on the error message to use downstream

**Comment by milocress** [comment]
I think it's cool that docformatter won't tell me which file is incorrect, it's like a fun little scavenger hunt.

File: llmfoundry/utils/exceptions.py (hunk 1 above)
Comment (jjanezhang) [reply llmfoundry/utils/exceptions.py:544]: Yea I think I'm going to rewrite to be something like "Your {train/eval} data path {table name} does not exist. Please double check your delta table name". If you could pass the delta table name we check as an attr on the class that would be great!

**Review by dakinggg — APPROVED** [review]
approving from foundry side. please make sure you and jane agree on the unresolved conversation before merging

**Pull request merged**